src/conf.py

```diff
@@ -1169,8 +1170,33 @@ def makeBanmask(self, hostmask, options=None, channel=None):
     similar.  It'll give up after it reads this many bytes, even if it hasn't
     found what it was looking for.""")))
 
+class HttpProxy(registry.String):
+  """Value must be a valid hostname:port string."""
+  def setValue(self, v):
+    if minisix.PY2:
+      from urllib2 import build_opener, install_opener, ProxyHandler
+    else:
+      from urllib.request import build_opener, install_opener, ProxyHandler
+    proxies = {}
+    if v != "":
+      # TODO: improve checks
+      if ':' not in v:
+        self.error()
+      try:
+        int(v.rsplit(':', 1)[1])
+      except ValueError:
+        self.error()
```

Comment:
There is probably a utility function somewhere (utils/ or urllib) to handle hostmask splitting.
Also, your code is broken if the user gives an IPv6 addresse without a port.